lib/ui/events.js

```diff
@@ -0,0 +1,317 @@
+      top           : 1,
+      width         : '100%',
+      height        : '50%-1',
+      border        : 'line',
+      align         : 'left',
+      keys          : true,
+      tags          : true,
+      mouse         : true,
+      noCellBorders : true,
+      scrollbar     : {
+        ch    : ' ',
+        style : { bg: 'white' },
+        track : {
+          style : { bg: 'grey' },
+        }
+      },
+      style : {
+        label  : { bold: true },
+        border : { fg: 'white' },
+        header : { fg: 'grey' },
+        cell   : { fg: 'white', selected: { bg: 'blue' } },
+      }
+    });
+    const event_description = blessed.box({
+      label         : 'Event description',
+      parent        : screen,
+      top           : '50%',
+      bottom        : '1',
+      width         : '100%',
+      height        : '50%-1',
+      border        : 'line',
+      align         : 'left',
+      keys          : true,
+      tags          : true,
+      mouse         : true,
+      noCellBorders : true,
+      scrollable    : true,
+      scrollbar     : {
+        ch    : ' ',
+        style : { bg: 'white' },
+        track : {
+          style : { bg: 'grey' },
+        }
+      },
+      style : {
+        label  : { bold: true },
+        border : { fg: 'white' },
+        header : { fg: 'grey' },
+        cell   : { fg: 'white', selected: { bg: 'blue' } },
+      }
+    });
+
+
+    // do not render when detached. 
+    events_table._orender = events_table.render;
+    events_table.render = function () {
+      if (this.parent == null) {
+        return;
+      }  
+      this._orender();
+    };
+
+    // work around to preserve selection and scrolling
+    events_table.on('detach', () => {
+      const { selected, childBase, childOffset } = events_table;
+      events_table.once('attach', () => {
+        events_table.select(selected);
+        Object.assign(events_table, { childBase, childOffset });
+        events_table.scrollTo(selected);
+      });
+    });
+
+    events_table.on('key C-z', () => {
+      cancellations.run(key);
+      navbar.remove(key);
+    });
+
+    const { promise, cancellation } = client.events(namespace).get({ cancellable : true });
+    cancellations.add(key, cancellation);
+    until(promise)
+    .spin(s => events_table.setLabel(`${s} ${key}`))
+    .succeed(_ => events_table.setLabel(`${key}`))
+    .then(response => {
+      const events_list = JSON.parse(response.body.toString('utf8'));
+      events_table.resource_version = events_list.metadata.resourceVersion;
+      this.fill_events_table(events_list, events_table, cancellations, key);
+    })
+    .then(_ =>{
+      const { promise, cancellation } = client.event(namespace)
+      .watch(events_table.resource_version)
+      .get({ generator: update_events_table.bind({ events_table, debug, highlightEvent })})
+      cancellations.add(key, cancellation);
+      return promise;
+    }).catch(error => {
```

Comment:
I think error should be caught in the `fail` method of the spinner otherwise it keeps spinning.